hi-ml/src/health_ml/model_trainer.py

```diff
@@ -189,22 +192,24 @@ def create_lightning_trainer(container: LightningContainer,
     return trainer, storing_logger
 
 
-def model_train(checkpoint_path: Optional[Path],
+def model_train(checkpoint_handler: Optional[CheckpointHandler],
                 container: LightningContainer,
                 num_nodes: int = 1) -> Tuple[Trainer, StoringLogger]:
     """
     The main training loop. It creates the Pytorch model based on the configuration options passed in,
     creates a Pytorch Lightning trainer, and trains the model.
     If a checkpoint was specified, then it loads the checkpoint before resuming training.
 
-    :param checkpoint_path: Checkpoint path for model initialization
+    :param checkpoint_handler: Checkpoint handler to retrieve model weights initialization and best checkpoint at the
```

Comment:
To be clearer here, can we say this is an instance of CheckpointHander rather than 'Checkpoint handler'. I know it sounds like nitpicking but someone reading this for the first time may not realise we have a class to create it